### ISSUE TITLE
CORS error using DICOMCloud with Cornerstone

### ISSUE THREAD

**Issue by shelbypereira** [DICOMcloud/DICOMcloud#57]
I have downloaded the Cornerstone library and am using it to test the DICOMCloud server.   I use the test page provided in the project cornerstoneWADOImageLoader/examples/wadors/index.html  and provide  a wadors query which hits the DICOMCLud server (http:localhost:44301/wadors/studies/2/series/2/instances/2 )
  
cornerstone then makes a call to  /wadors/studies/2/series/2/instances/2/metadata which is serviced by DICOMCloud correctly

however the next call is /wadors/studies/2/series/2/instances/2/metadata/frames/1

and this call when viewed in the network panel of the Chrome debugger shows a CORS error:
Access to XMLHttpRequest at 'http://localhost:44301/wadors/studies/2/series/2/instances/2/metadata/frames/1' from origin 'null' has been blocked by CORS policy: Request header field accept is not allowed by Access-Control-Allow-Headers in preflight response.

I checked a bit on internet and it seems some accept-headers may be missing as described in SO here:
[https://stackoverflow.com/questions/32500073/request-header-field-access-control-allow-headers-is-not-allowed-by-itself-in-pr
]

I will try to add the accept headers or disable CORS,  any thoughts ?

**Comment by dannyrb** [comment]
`DICOMCloud` should not have wildcard accept headers by default. If you're requesting resources from a DICOMCloud instance from web page hosted at a different domain, you'll need to make sure it returns appropriate CORS Headers. 

IIS supports custom headers, or the application can add them. If you're using IIS, some older versions require you to modify the handler mappings to allow the site to handle OPTIONS requests.

I'm only familiar with [how you would go about doing this in .NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/cors?view=aspnetcore-2.2), but others here may be of more help. It's not uncommon to make the CORS options configurable with environment variables.

**Comment by Zaid-Safadi** [comment]
So DICOMcloud enabled CORS by default while it is still in beta but the problem is in the OPTIONS PreFlight Request which is sent by the browser for WADO-RS requests. 

The OPTIONS PreFlight requires special handling and it is a pain in .NET Web API 2 to work.
I still need to add some documentation but I have it implemented [here](https://github.com/DICOMcloud/DICOMcloud/commit/7bd7a80eed323d56675c5101442d978a573874b8)

Bottom line, now from the web.config you have these options:
```
    <add key="cors:enabled" value="true"/>
    <add key="cors:origins" value="*"/>
    <add key="cors:headers" value="*"/>
    <add key="cors:methods" value="*"/>
```

**Comment by Zaid-Safadi** [comment]
This is now in Development and enabled by default
